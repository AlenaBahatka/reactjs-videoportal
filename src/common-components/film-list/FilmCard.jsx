import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router-dom';

FilmCard.propTypes = {
	title: PropTypes.string,
	director: PropTypes.string,
	year: PropTypes.number
};

function FilmCard(props) {
	let { title, director, year, coverPicture, filmId } = props;
	const uri = `/film/${filmId}`;

	return (
		<div className="col-sm-4">
			<img src={coverPicture} alt={title} />
			<Link to={uri}>
				<h2>{title}</h2>
			</Link>
			<span> {director} </span>
			<span> {year} </span>
		</div>
	);
}

export default withRouter(FilmCard);
