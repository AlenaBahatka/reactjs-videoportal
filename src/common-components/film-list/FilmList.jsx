import PropTypes from 'prop-types';
import React from 'react';
import FilmCard from './FilmCard';

FilmList.propTypes = {
	films: PropTypes.array
};

function FilmList(props) {
	return (
		<div>
			{props.films.map((film, i) => (
				<FilmCard
					title={film.title}
					director={film.director}
					year={film.year}
					coverPicture={film.coverPicture}
					key={film.id}
				/>
			))}
		</div>
	);
}

export default FilmList;
