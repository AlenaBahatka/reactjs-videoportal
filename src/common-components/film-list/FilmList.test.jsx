import React from 'react';
import { shallow, mount } from 'enzyme';
import FilmCard from './FilmCard';
import FilmList from './FilmList';

let films;
describe('FilmList', () => {
	it('should correctly render component in case array is empty', () => {
		films = [];
		let filmListComponent = shallow(<FilmList films={films} />);

		expect(filmListComponent.find('div')).toBeTruthy();
		expect(filmListComponent.find('div').children().length).toBeFalsy();
	});

	describe('FilmList - some films in array', () => {
		beforeAll(() => {
			films = [
				{
					title: 'Roma',
					director: 'Alfonso Cuarón',
					year: 2017,
					coverPicture: 'pic.jpg',
					id: 'id1'
				},
				{
					title: 'Tristana',
					director: 'Luis Buñuel',
					year: 1970,
					coverPicture: 'pic2.jpg',
					id: 'id2'
				}
			];
		});

		it('should correctly render 2 items', () => {
			let filmListComponent = shallow(<FilmList films={films} />);

			expect(filmListComponent).toMatchSnapshot();
		});

		it('should correctly render 2 items - FilmCard', () => {
			let filmListComponent = mount(<FilmList films={films} />);

			expect(filmListComponent.find(FilmCard).length).toBe(2);
		});
	});
});
