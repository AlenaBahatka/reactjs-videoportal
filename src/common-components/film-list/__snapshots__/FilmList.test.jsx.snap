// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList FilmList - some films in array should correctly render 2 items 1`] = `
<div>
  <FilmCard
    coverPicture="pic.jpg"
    director="Alfonso Cuarón"
    key="id1"
    title="Roma"
    year={2017}
  />
  <FilmCard
    coverPicture="pic2.jpg"
    director="Luis Buñuel"
    key="id2"
    title="Tristana"
    year={1970}
  />
</div>
`;
